File: common/src/main/java/com/ssblur/scriptor/word/action/teleport/BringAction.java
package com.ssblur.scriptor.word.action.teleport;

import com.ssblur.scriptor.helpers.targetable.Targetable;
import com.ssblur.scriptor.api.word.Descriptor;

public class BringAction extends SwapAction {
  @Override
  public void apply(Targetable caster, Targetable targetable, Descriptor[] descriptors) {
    if(targetable.getLevel().isClientSide
      || caster == null
      || caster.getLevel() == null
      || targetable.getLevel() == null) return;

    teleport(targetable, caster);
  }

  @Override
  public Cost cost() { return new Cost(6, COSTTYPE.ADDITIVE); }

}

File: common/src/main/kotlin/com/ssblur/scriptor/word/action/teleport/BringAction.kt
package com.ssblur.scriptor.word.action.teleport

import com.ssblur.scriptor.api.word.Descriptor
import com.ssblur.scriptor.helpers.targetable.Targetable

class BringAction : SwapAction() {
    override fun apply(caster: Targetable, targetable: Targetable, descriptors: Array<Descriptor>) {
        if (targetable.level.isClientSide) return

        teleport(targetable, caster)
    }

    override fun cost(): Cost {
        return Cost(6.0, COSTTYPE.ADDITIVE)
    }
}
